### DISCUSSION TITLE
How to add /*#__PURE__/ (or any comment) to a call expression in swc plugin

### DISCUSSION THREAD

**Discussion by rishabh3112** [swc-project/swc#8423] [Q&A] [answered]
I am working on graphql-tag-swc-plugin, where I have to add a pure function call expression to the `Program`.
I am done with addition but, I need to add leading comment to the call expression so that terser / minimiser can treat it as pure. After looking through docs, unable to find a way to add comments.

I found a trait for this: https://rustdoc.swc.rs/swc_common/comments/trait.Comments.html#tymethod.add_pure_comment
But don't know how to implement it.

**Comment by kdy1** [comment, marked as answer]
https://github.com/swc-project/plugins/tree/6dca40dc542fda2bac9ea2976793326d190f80ab/packages/swc-magic

**Reply by rishabh3112** [reply]
Thanks @kdy1
